140-runner-host-mounts/proposal.md

```diff
@@ -0,0 +1,76 @@
+# Summary
+
+This proposal introduces a mechanism through which tasks can request files, directories, or
+block devices to be mounted from the host the worker is on to the task's container.
+
+The worker must be configured to allow tasks to mount a given location, which we will make clear
+is a last-resort option when other mechanisms are unsuitable.
+
+# Motivation
+
+When performing tasks that require access to a device on the host, such as a GPU, it is required to
+make the 'files' that the kernel creates available in a task's container.
+
+For instance, to use an AMD GPU within a container, the container needs access to these paths from the host:
+
+  - `/dev/kfd` - GPU compute interface
+  - `/dev/dri` - folder containing an interface for each GPU.
+
+Currently, it is not possible to run these workloads using Concourse.
+
+# Proposal
+
+## Worker configuration option `--allowed-host-mounts`
+
+Add a new configuration option to the worker command, `--allowed-host-mounts`:
+The value should be a regular expression. Any host path that the expression fully matches will be *allowed* to be
+mounted into a task's container if it is requested.
+
+This option can be specified only once, but can use the regex OR operator `|` to match separate paths as required.
```

Comment:
Might be better to use a list of regexes, otherwise it might get ugly if you need many different paths.